bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/InvalidDataException.java

```diff
@@ -0,0 +1,17 @@
+package org.openhab.binding.myq.internal;
+
+import java.io.IOException;
+
+/**
+ * Throw if the data we are parsing in not what we are expecting for input. 
+ * @author Dan Cunningham
+ *
+ */
+public class InvalidDataException extends IOException {
```

Comment:
This is how the Jackson json parser implements its exceptions ( like JsonParseException) if the JSON is malformed or unexpected. I liked this pattern and thought it fit this use case well as the developer can be very specific about the exception they want to catch or not.